packages/compass-e2e-tests/helpers/commands/select-stage-operator.ts

```diff
@@ -1,32 +1,40 @@
 import type { CompassBrowser } from '../compass-browser';
+import delay from '../delay';
 import * as Selectors from '../selectors';
+import { focusStageOperator } from './focus-stage-operator';
 
 export async function selectStageOperator(
   browser: CompassBrowser,
   index: number,
   stageOperator: string
 ): Promise<void> {
-  const inputSelector = Selectors.stageSelectControlInput(index);
+  const comboboxSelector = Selectors.stagePickerComboboxInput(index);
   const textareaSelector = Selectors.stageTextarea(index);
 
-  // it should become focused straight after focusStageSelector()
+  await focusStageOperator(browser, index);
+
+  await browser.setValueVisible(comboboxSelector, stageOperator);
+
+  await browser.keys(['Enter']);
+
+  // the "select" should now blur and the ace textarea become focused
   await browser.waitUntil(async () => {
-    const inputElement = await browser.$(inputSelector);
+    const inputElement = await browser.$(comboboxSelector);
     const isFocused = await inputElement.isFocused();
-    return isFocused === true;
+    return isFocused === false;
   });
 
-  await browser.setValueVisible(inputSelector, stageOperator);
-  await browser.keys(['Enter']);
-
   // click the textarea to focus the stage (this should happen automatically,
   // but flakes very often in CI)
   await browser.$(textareaSelector).click();
 
-  // the "select" should now blur and the ace textarea become focused
   await browser.waitUntil(async () => {
     const textareaElement = await browser.$(textareaSelector);
     const isFocused = await textareaElement.isFocused();
     return isFocused === true;
   });
+
+  // NOTE: even if the focus moved away from the combobox interactions are still
+  // targeting the combobox
+  await delay(1000);
```

Comment:
@lerouxb any suggestion to avoid this here?

Reply:
wait for something (like the options popover thingy) to not be displayed or exist (see our usage of reverse: true) or wait for something to be not focused in a waitUntil?

Reply:
I don't think this is used in the time to first query test, but anywhere it comes up where we might measure the time these delays will skew results. sometimes it is pretty hard to avoid, though.

Reply:
is so strange, it is like there is some event listener still attached for a while after the dropdown is hidden, I'm already waiting for the focus to go somewhere else and for the dropdown to be hidden, no idea what else I can wait for, I'll check the `{reverse: true}`

Reply:
waitForExist({ reverse: true }) on something, maybe?